.github/workflows/pre-commit.yml

```
@@ -0,0 +1,24 @@
+name: Pre-commit checks
+
+on:
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v5
+
+      - name: Install required tools
+        run: |
+          pip install pre-commit
+          make bootstrap
+          go mod tidy && go mod vendor
```

we're not using vendoring, so this should not be necessary
```suggestion
          go mod tidy
```